src/user/views/user_views.py

```diff
@@ -124,6 +124,29 @@ def referral_rsc(self, request):
 
         return Response({"amount": amount})
 
+    @action(
+        detail=True,
+        methods=[RequestMethods.POST],
+        permission_classes=[RequestorIsOwnUser],
+    )
+    def set_should_display_rsc_balance(self, request, pk=None):
+        try:
+            user = User.objects.get(id=request.user.id)
```

Comment:
Is the request coming as `api/user/{user_id}/set_should_display_rsc_balance`? If so, it can be `user = self.get_object()`

Reply:
i want the FE payload to specify which user is the target user for the endpoint. 
Permission check is already being done.